src/abstract/WorldIDBridge.sol

```diff
@@ -131,21 +131,21 @@ abstract contract WorldIDBridge is IWorldID {
     /// @custom:reverts ExpiredRoot If the provided `root` has expired.
     /// @custom:reverts NonExistentRoot If the provided `root` does not exist in the history.
     function checkValidRoot(uint256 root) public view returns (bool isValid) {
```

Comment:
This API is horrible in general – this method either returns true or reverts. It should do one of these things  – i.e. either `requireValidRoot` that just reverts or an `isValidRoot` that returns a bool. The current state is very error prone and awkward (as evidenced by your newly added `unreachable` branch).